editor/sidebar/post-schedule/index.js

```diff
@@ -0,0 +1,102 @@
+/**
+ * External dependencies
+ */
+import { connect } from 'react-redux';
+import clickOutside from 'react-click-outside';
+import DatePicker from 'react-datepicker';
+import moment from 'moment';
+
+/**
+ * WordPress dependencies
+ */
+import { __ } from 'i18n';
+import { Component } from 'element';
+import { dateI18n, settings } from 'date';
+
+/**
+ * Internal dependencies
+ */
+import './style.scss';
+import PostScheduleClock from './clock';
+import { getEditedPostAttribute } from '../../selectors';
+import { editPost } from '../../actions';
+
+class PostSchedule extends Component {
+	constructor() {
+		super( ...arguments );
+		this.state = {
+			open: false,
+		};
+		this.toggleDialog = this.toggleDialog.bind( this );
+	}
+
+	toggleDialog( event ) {
+		event.preventDefault();
+		this.setState( { opened: ! this.state.opened } );
+	}
+
+	handleClickOutside() {
+		this.setState( { opened: false } );
+	}
+
+	render() {
+		const { date, onUpdateDate } = this.props;
+		const momentDate = date ? moment( date ) : moment();
+		const label = date
+			? dateI18n( settings.formats.datetime, date )
+			: __( 'Immediately' );
+		const handleChange = ( newDate ) => {
+			onUpdateDate( newDate.format( 'YYYY-MM-DDTHH:mm:ss' ) );
+		};
+
+		// To know if the current timezone is a 12 hour time with look for "a" in the time format
+		// We also make sure this a is not escaped by a "/"
+		const is12HourTime = /a(?!\\)/i.test(
+			settings.formats.time
+				.toLowerCase() // Test only the lower case a
+				.replace( /\\\\/g, '' ) // Replace "//" with empty strings
+				.split( '' ).reverse().join( '' ) // Reverse the string and test for "a" not followed by a slash
+		);
```

Comment:
I know how old this PR is :), but today we have multiple occurrences of this snippet in the codebase, so it's worth asking: 

* Why reverse the string and not use a non-capturing group like `/(?:[^\\])a/`?
* If the RegExp already has the `i` flag, why pipe through `.toLowerCase`?